app/command_autocomplete.go

```diff
@@ -174,18 +174,39 @@ func (a *App) parseArgument(commandArgs *model.CommandArgs, arg *model.Autocompl
 		}
 		parsed = changedParsed
 		toBeParsed = changedToBeParsed
+	} else if arg.Type == model.AutocompleteArgTypeBool {
+		found, changedParsed, changedToBeParsed, boolsuggestions := parseBoolArgument(arg, parsed, toBeParsed)
+		if found {
+			suggestions = append(suggestions, boolsuggestions)
+			return true, changedParsed, changedToBeParsed, suggestions
+		}
+		parsed = changedParsed
+		toBeParsed = changedToBeParsed
 	}
 
 	return false, parsed, toBeParsed, suggestions
 }
 
 func parseNamedArgument(arg *model.AutocompleteArg, parsed, toBeParsed string) (found bool, alreadyParsed string, yetToBeParsed string, suggestion model.AutocompleteSuggestion) {
 	in := strings.TrimPrefix(toBeParsed, " ")
+	if len(arg.Name) == 0 && arg.Type == model.AutocompleteArgTypeBool {
+		if len(parsed) > 0 || len(toBeParsed) > 0 {
+			return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed + " " + "false" + " ", Suggestion: "false", Hint: "", Description: arg.HelpText}
+		}
+		return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: "false" + " ", Suggestion: "false", Hint: "", Description: arg.HelpText}
+	}
+
 	namedArg := "--" + arg.Name
 	if in == "" { //The user has not started typing the argument.
+		if arg.Type == model.AutocompleteArgTypeBool {
+			return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed + namedArg + " " + "true" + " ", Suggestion: "true", Hint: "", Description: arg.HelpText}
+		}
 		return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed + namedArg + " ", Suggestion: namedArg, Hint: "", Description: arg.HelpText}
 	}
 	if strings.HasPrefix(strings.ToLower(namedArg), strings.ToLower(in)) {
+		if arg.Type == model.AutocompleteArgTypeBool {
+			return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed + namedArg[len(in):] + " " + "true" + " ", Suggestion: "true", Hint: "", Description: arg.HelpText}
```

Comment:
same here, when bool type argument is named we do not need `true` and `false`, just `--argname` is enough